der/src/reader/pem.rs

```diff
@@ -51,9 +48,11 @@ impl<'i> Reader<'i> for PemReader<'i> {
         self.position.input_len()
     }
 
-    fn peek_into(&self, buf: &mut [u8]) -> Result<()> {
-        self.clone().read_into(buf)?;
-        Ok(())
+    fn peek<T, F>(&self, f: F) -> Result<T>
+    where
+        F: FnOnce(&mut Self) -> Result<T>,
+    {
+        f(&mut self.clone())
     }
```

Comment:
Really this is wrapping up cloning the `Reader`. We could potentially just bound `Reader` on `Clone` instead.